src/string.rs

```diff
@@ -456,4 +476,62 @@ mod test {
         assert_eq!(core::mem::size_of::<FixedStringRepr<u32>>(), 13);
         assert_eq!(core::mem::align_of::<FixedStringRepr<u32>>(), 1);
     }
+
+    #[test]
+    fn from_char_u8() {
+        let s: FixedString<u8> = 'a'.into();
+        assert_eq!(s.len(), 1);
+        assert!(s.is_inline());
+
+        let s: FixedString<u8> = '¼'.into();
+        assert_eq!(s.len(), 2);
+        assert!(s.is_inline());
+
+        let s: FixedString<u8> = '⚡'.into();
+        assert_eq!(s.len(), 3);
+        assert!(s.is_inline());
+
+        let s: FixedString<u8> = '🦀'.into();
+        assert_eq!(s.len(), 4);
+        assert!(s.is_inline());
```

Comment:
This will fail on 16 bit, right? Since there will only be space for a 2 byte pointer and a 1 byte length value, and we need to fit 4 bytes in.

If so, can you `cfg` this `is_inline` assertion for only 32 bit and above? If I am wrong here, can you remove the heap spilling logic from the `From<char>` impl since it should never be hit if this assert always passes.

Reply:
Yes, you are right. Will fail on 16 bit.